WordPress/gutenberg
# Lodash: Refactor away from `_.escape()`

## What?
This PR removes Lodash's `_.escape()` completely and deprecates the method. There are just a few usages and conversion is pretty straightforward. 

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

We're replacing the single usage with the `escape-html` package that does exactly the same thing and is widely used throughout the JS world.

## Testing Instructions
* Verify the Navigation block, creating and editing items still works well.
* Verify you can still insert an existing tag and create a new tag for a post properly.
* Verify all checks are green and all tests still pass.
